Watchman.Discord/Areas/Protection/Controllers/MuteUserController.cs

```diff
@@ -13,32 +19,46 @@ public class MuteUserController : IController
         private readonly MutingService _mutingService;
         private readonly UnmutingService _unmutingService;
         private readonly UsersService _usersService;
+        private readonly MessagesServiceFactory _messagesServiceFactory;
 
-        public MuteUserController(MutingService mutingService, UnmutingService unmutingService, UsersService usersService)
+        public MuteUserController(MutingService mutingService, UnmutingService unmutingService, UsersService usersService, MessagesServiceFactory messagesServiceFactory)
         {
             this._mutingService = mutingService;
             this._unmutingService = unmutingService;
             this._usersService = usersService;
+            this._messagesServiceFactory = messagesServiceFactory;
         }
 
-        [DiscordCommand("mute")]
         [AdminCommand]
-        public async Task MuteUser(DiscordRequest request, Contexts contexts)
+        public async Task MuteUser(MuteCommand command, Contexts contexts)
         {
-            var requestParser = new MuteRequestParser(request, this._usersService, contexts);
-            var userToMute = requestParser.GetUser();
-            var muteEvent = requestParser.GetMuteEvent(userToMute.Id, contexts, request);
+            var requestParser = new MuteRequestParser(this._usersService, contexts);
+            var userToMute = await requestParser.GetUser(command.Mention);
+            MuteEvent? muteEvent = null;
+            try
+            {
+                muteEvent = requestParser.GetMuteEvent(userToMute.Id, contexts, command.Reason, command.Time);
+            }
+            catch (InvalidArgumentsException)
+            {
+                var messagesService = this._messagesServiceFactory.Create(contexts);
+                await messagesService.SendResponse(x => x.InvalidArguments());
+            }
+            catch (TimeIsTooBigException)
+            {
+                var messagesService = this._messagesServiceFactory.Create(contexts);
+                await messagesService.SendResponse(x => x.TimeIsTooBig());
```

Comment:
to powinno być ogarniane na poziomie frameworka 
mamy wyjątki które są łapane przez framework, który odsyła odpowiednie odpowiedzi

poza tym po wyjątku chcielibyśmy nie wywoływać dalej komendy
a u ciebie jest
```csharp
await this._mutingService.MuteUserOrOverwrite(contexts, muteEvent, userToMute);
this._unmutingService.UnmuteInFuture(contexts, muteEvent, userToMute);
```

Reply:
w sensie, że jak na poziomie frameworka? To gdzie to ma być?